pendulum_control/include/pendulum_control/rtt_executor.hpp

```diff
@@ -52,31 +52,22 @@ class RttExecutor : public rclcpp::executor::Executor
   /// Default destructor
   virtual ~RttExecutor() {}
 
-  /// Return the status of rttest.
-  // \return True if rttest has initialized, false if it is uninitialized or has finished.
-  bool is_rttest_ready() const
-  {
-    return rttest_ready;
-  }
-
   /// Return true if the executor is currently spinning.
   // \return True if rclcpp is running and if the "running" boolean is set to true.
   bool is_running() const
   {
     return rclcpp::ok() && running;
   }
 
-  /// Retrieve the results measured by rttest
-  // \param[in] output A struct containing performance statistics.
-  void get_rtt_results(rttest_results & output) const
-  {
-    output = results;
-  }
-
-  void set_rtt_results_message(pendulum_msgs::msg::RttestResults & msg) const
+  bool set_rtt_results_message(pendulum_msgs::msg::RttestResults & msg) const
   {
+    if (!results_available) {
+      return false;
+    }
     msg.cur_latency = last_sample;
     msg.mean_latency = results.mean_latency;
+    assert(results.min_latency >= 0);
```

Comment:
I guess it's not hurting to have these assertions here, but it seems superfluous since `min_latency` and `max_latency` are non-negative by declaration:

https://github.com/ros2/demos/blob/97651912ebad439a6bc62c34671cad6a5bfee785/pendulum_msgs/msg/RttestResults.msg#L8-L9

Reply:
It's a little complicated, but the underlying code in https://github.com/ros2/realtime_support/blob/aef9318be62e1b63753d55daab9606510f0b9c46/rttest/src/rttest.cpp actually allows for negative numbers.  The idea with the assertions here is to make it obvious (via a test failure) if we ever try to stuff a negative number into the unsigned 64-bit quantity.  If we remove these, then it could lead to a difficult-to-figure bug where we see a giant number in the unsigned field.  It's a somewhat arbitrary distinction (it's a bug either way), but this just makes it a bit more obvious.  Does that make sense?

Reply:
Nevermind, I my brain parsed this as asserting on the message type. I see now `results` has type `rttest_results`. This makes sense.